src/circomkit.ts

```diff
@@ -404,22 +426,22 @@ export class Circomkit {
       if (this.config.prime !== 'bn128') {
         throw new Error('Please provide PTAU file when using a prime field other than bn128');
       }
-      this.log('Checking for PTAU file...', 'debug');
-      ptauPath = await this.ptau(circuit);
     }
 
+    // check if provided ptau actually exists
+    // or download if not provided
+    this.log('Checking for PTAU file...', 'debug');
+    ptauPath = await this.ptau(circuit);
```

Comment:
Shouldn't this be inside the `if (ptauPath === undefined)` check? Otherwise, it will overwrite the `ptauPath` optional parameter.

Reply:
ah nvm, `ptau` function does that check hahah